presto-native-execution/src/test/java/com/facebook/presto/spark/TestPrestoSparkNativeSimpleQueries.java

```diff
@@ -166,4 +169,34 @@ public void testAggregationCompanionFunction()
             getQueryRunner().execute("DROP TABLE IF EXISTS avg_partial_states");
         }
     }
+
+    @Test
+    public void testRetryOnOutOfMemoryBroadcastJoin()
+    {
+        String query = "select l.orderkey from lineitem l join orders o on l.orderkey = o.orderkey ";
+
+        Session session = Session.builder(getSession())
+                .setSystemProperty(JOIN_DISTRIBUTION_TYPE, "BROADCAST")
+                .setSystemProperty(SPARK_BROADCAST_JOIN_MAX_MEMORY_OVERRIDE, "10B")
+                .setSystemProperty(SPARK_RETRY_ON_OUT_OF_MEMORY_BROADCAST_JOIN_ENABLED, "false")
+                .build();
+
+        // Query should fail with broadcast join OOM
+        assertQueryFails(
+                session,
+                query,
+                "Query exceeded per-node broadcast memory limit of 10B \\[Compressed broadcast size: .*kB\\]");
+
+        Session expectedSession = Session.builder(getSession())
+                .setSystemProperty(JOIN_DISTRIBUTION_TYPE, "BROADCAST")
+                .build();
+        Session actualSession = Session.builder(getSession())
```

Comment:
Consider extracting a helper method with a boolean parameter to create a session with broadcast limit and retry enabled/disabled to avoid copy-paste.